src/shared/ui/input.ts

```diff
@@ -0,0 +1,128 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as vscode from 'vscode'
+
+/**
+ * Options to configure the behavior of the input box UI.
+ * Generally used to accommodate features not provided through vscode.InputBoxOptions
+ */
+export interface AdditionalInputBoxOptions {
+    title?: string
+}
+
+/**
+ * Creates an InputBox to get a text response from the user.
+ *
+ * Used to wrap createInputBox and accommodate
+ * a common set of features for the Toolkit.
+ *
+ * Parameters:
+ *  options - initial InputBox configuration
+ *  buttons - set of buttons to initialize the InputBox with
+ * @return A new InputBox.
+ */
+export function createInputBox({
+    options,
+    buttons,
+}: {
+    options?: vscode.InputBoxOptions & AdditionalInputBoxOptions,
+    buttons?: vscode.QuickInputButton[],
+}): vscode.InputBox {
+
+    const inputBox = vscode.window.createInputBox()
+
+    if (options) {
+        inputBox.title = options.title
+        inputBox.placeholder = options.placeHolder
+        inputBox.prompt = options.prompt
+        if (options.ignoreFocusOut) { inputBox.ignoreFocusOut = options.ignoreFocusOut }
```

Comment:
This won't set `inputBox.ignoreFocusOut` if `options.ignoreFocusOut` is set to `false`. Is this intentional? If not, explicitly compare with `undefined` instead of checking truthiness.

Reply:
Good catch, the intention was "if not undefined"